src/Utils.cc

```diff
@@ -652,17 +652,17 @@ void radix_sort_kernel(
     const V* const input_values,
     K* const output_keys,
     V* const output_values,
-    const int elements_count,
-    int* const histogram,
-    int* const histogram_ps,
+    const int64_t elements_count,
+    int64_t* const histogram,
+    int64_t* const histogram_ps,
     const int pass,
     const bool pass_with_sign_bit = false) {
   const auto tid = omp_get_thread_num();
   const auto nthreads = omp_get_num_threads();
   const auto elements_count_4 = elements_count / 4 * 4;
 
-  int* const local_histogram = &histogram[RDX_HIST_SIZE * tid];
-  int* const local_histogram_ps = &histogram_ps[RDX_HIST_SIZE * tid];
+  int64_t* const local_histogram = &histogram[RDX_HIST_SIZE * tid];
```

Comment:
Can we use `auto` here so that we won't have to worry about typing?

Reply:
Done. Though I would like to emphasize the difference between:
`const auto ptr = &val;  // equivalent to e.g. int64_t* const ptr = &val`
and
`const auto* ptr = &val; // equivalent to e.g. const int64_t* ptr = &val;`
I hope that code readers will not be confused by auto usage here.